lib/galaxy/managers/workflows.py

```diff
@@ -1059,13 +1079,19 @@ def _workflow_to_dict_run(self, trans, stored, workflow, history=None):
                 }
                 for oc in step.output_connections
             ]
+            annotations_dict = {
+                "annotation": stored.annotations[0].annotation,
+                "update_time": (stored.annotations[0].stored_workflow.update_time).isoformat(),
```

Comment:
```suggestion
                "update_time": (stored.update_time).isoformat(),
```
However you're in the step loop, so you're doing this once per step. There's also no guarantee that there is at least one annotation, so `stored.annotations[0].annotation` will fail often.

Reply:
Thank you! This answers my question.